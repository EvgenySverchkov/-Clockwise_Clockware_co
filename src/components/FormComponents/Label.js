import React from "react";
import PropTypes from "prop-types";

const Label = ({ children, forId, isFontWeight=true }) => (
  <label
    htmlFor={forId}
    className={`pl-0 pr-0 col-sm-4 col-md-4 col-form-label ${isFontWeight ? "font-weight-bold": ''}`}
  >
    {children}
  </label>
);

Label.propTypes = {
  children: PropTypes.node.isRequired, 
  forId: PropTypes.string.isRequired, 
  isFontWeight: PropTypes.bool
}

export default Label;
