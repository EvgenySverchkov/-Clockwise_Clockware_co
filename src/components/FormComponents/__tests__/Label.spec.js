import React from 'react';
import {create} from "react-test-renderer";

import Label from "../Label";

it("<Label /> test which is used in forms", ()=>{
    const mockProps = { 
        forId: "1"
    }
    const component = create(<Label {...mockProps}>
        <div>Some element</div>
    </Label>)
    expect(component.toJSON()).toMatchSnapshot();
});