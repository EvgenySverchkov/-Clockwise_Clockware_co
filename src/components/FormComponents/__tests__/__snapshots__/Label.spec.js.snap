// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> test which is used in forms 1`] = `
<label
  className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label font-weight-bold"
  htmlFor="1"
>
  <div>
    Some element
  </div>
</label>
`;
